avocado/core/nrunner.py

```diff
@@ -664,6 +665,7 @@ class Task:
     def __init__(self, identifier, runnable, status_uris=None,
                  known_runners=None):
         self.identifier = identifier
+        self.uid = str(uuid4())
```

Comment:
A unique `uuid4` is a good idea here. But this might be a little confusing since we have already an `identifier`.

I would recommend change the API here and rename the identifier with `name` or something.

Reply:
I agree with you, and my first intention was to change the API.

When I started evaluating the change, I noticed it would require a massive change in the code. This would impact the current command line syntax we have.

It is worth changing it, but on a separate PR, not related to the requirements resolver.

Reply:
In this case it might be worth noting it down for everyone in an issue as a TODO card that could be tracked globally.

Reply:
Sure, as soon as this work is merged, I'll open the issue.